src/Core/Services/Implementations/HandlebarsMailService.cs

```diff
@@ -19,17 +19,20 @@ public class HandlebarsMailService : IMailService
 
         private readonly GlobalSettings _globalSettings;
         private readonly IMailDeliveryService _mailDeliveryService;
+        private readonly IMailEnqueuingService _mailEnqueuingService;
         private readonly Dictionary<string, Func<object, string>> _templateCache =
             new Dictionary<string, Func<object, string>>();
 
         private bool _registeredHelpersAndPartials = false;
 
         public HandlebarsMailService(
             GlobalSettings globalSettings,
-            IMailDeliveryService mailDeliveryService)
+            IMailDeliveryService mailDeliveryService,
+            IMailEnqueuingService mailEnqueuingService)
```

Comment:
Should we consider an `AzureQueueMailService` to make all mail go through this queueing strategy?

Reply:
I would like to maintain a single location for building handlebars models. Maybe an abstract `MailQueueService` that implements those things and an `AzureQueueMailService` and `BlockingMailService`. `HandlebarsMailService` would be reduced to processing `MailQueueMessage`s.

It's just a much larger body of work to re-implement every email as a queue. If you're willing to review and we can QA test it, I'm willing to write it 😄

Reply:
What about creating a `AzureQueueMailDeliveryService`? Instead of sending the mail, it just queues it. That way you can still use the `HandlebarsMailService` as-is.

Reply:
Do you mean enqueue the completed message? I didn't do that because I'm worried about the full length of our emails. Enqueuing a model and template name is always going to be shorter than the full rendered message.

For Org invites, it's 1.3kb message vs 21.2kb. That should translate to ~20x slower to enqueue a full email vs the min data set.

Reply:
Hmm, yea, that's no good. This is probably fine then.